cmd/mongodb-healthcheck/main.go

```diff
@@ -128,14 +141,18 @@ func main() {
 
 		case "mongod":
 			log.Error("readiness check for mongod is not implemented")
-			session.Close()
+			if err := client.Disconnect(context.TODO()); err != nil {
+				log.Errorf("failed to disconnect: %v", err)
+			}
 			os.Exit(1)
 
 		case "mongos":
-			err := healthcheck.MongosReadinessCheck(session)
+			err := healthcheck.MongosReadinessCheck(client)
 			if err != nil {
 				log.Error(err.Error())
-				session.Close()
+				if err := client.Disconnect(context.TODO()); err != nil {
+					log.Errorf("failed to disconnect: %v", err)
```

Comment:
maybe it's better to put  something like 'errorMessage' into `if` and after it call log.Errorf(...

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/757/commits/b7032b2da10a4b4eb30909772ea64bb56beb9f33